Samples/Notifications/App/CsUnpackagedAppNotifications/CsUnpackagedAppNotifications/NotificationManager.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.Windows.AppNotifications;
+using Windows.Foundation;
+using Microsoft.UI.Xaml.Controls;
+
+namespace CsUnpackagedAppNotifications
+{
+    internal class NotificationManager
+    {
+        private bool m_isRegistered;
+
+        private Dictionary<string, Action<AppNotificationActivatedEventArgs>> c_map;
+
+        public NotificationManager()
+        {
+            m_isRegistered = false;
+
+            c_map = new Dictionary<string, Action<AppNotificationActivatedEventArgs>>();
+            c_map.Add("1", ToastWithAvatar.NotificationReceived);
+            c_map.Add("2", ToastWithTextBox.NotificationReceived);
+        }
+
+        ~NotificationManager()
+        {
+            Unregister();
+        }
+
+        public void Init()
+        {
+            AppNotificationManager notificationManager = AppNotificationManager.Default;
+
+            notificationManager.NotificationInvoked += OnNotificationInvoked;
+
+            notificationManager.Register();
```

Comment:
Maybe this would be a good place for: 
```
            // To ensure all Notification handling happens in this process instance, register for
            // NotificationInvoked before calling Register(). Without this a new process will
            // be launched to handle the notification.
```

Reply:
Done